### DISCUSSION TITLE
uniform kernel

### DISCUSSION THREAD

**Discussion by ekinugurel** [cornellius-gp/gpytorch#2208] [Q&A]
Hi, relatively new GPyTorch user here. I am looking to implement a constant (uniform) kernel in the following form
![image](https://user-images.githubusercontent.com/57807000/204671540-af5a38e6-7757-4f7a-8300-c52e440ec455.png)

I took the example sinc kernel shown in [here](https://docs.gpytorch.ai/en/stable/examples/00_Basic_Usage/Implementing_a_custom_Kernel.html) and tried something similar using np.piecewise. 

`class ConstantKernel(gpytorch.kernels.Kernel):
    has_lengthscale = True

    # this is the kernel function
    def forward(self, x1, x2, **params):
        # apply lengthscale
        x1_ = x1.div(self.lengthscale)
        x2_ = x2.div(self.lengthscale)
        # calculate the distance between inputs
        diff = self.covar_dist(x1_, x2_, **params)
        # prevent divide by 0 errors
        diff.where(diff == 0, torch.as_tensor(1e-20))
        # return sinc(diff) = sin(diff) / diff
        return np.piecewise(diff, [diff <= -(self.lengthscale), 
                                   abs(diff) <= self.lengthscale, 
                                   diff >= self.lengthscale ], 
                            [0, 0.5*(diff+1), 1])`

But it doesn't seem to work. Can anyone advise on this approach? Sorry for the simple question.

**Comment by gpleiss** [comment]
What do you mean by "it doesn't seem to work"? Is it throwing errors? Returning incorrect results?

One thing of the top of my head - don't use `np.piecewise`. Numpy and torch functions don't always play well together.
